spring/boot1-autoconfigure/src/main/java/com/linecorp/armeria/spring/DurationStyle.java

```diff
@@ -0,0 +1,268 @@
+ * limitations under the License.
+ */
+package com.linecorp.armeria.spring;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.springframework.util.Assert;
+import org.springframework.util.StringUtils;
+
+/**
+ * Duration format styles.
+ */
+enum DurationStyle {
+
+    // Forked from Spring Boot 2.3.1.
+    // https://github.com/spring-projects/spring-boot/blob/b21c09020da7b237731d69c5c96b163e810c200f/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/convert/DurationStyle.java
+
+    /**
+     * Simple formatting, for example '1s'.
+     */
+    SIMPLE("^([+-]?\\d+)([a-zA-Z]{0,2})$") {
+        @Override
+        Duration parse(String value, ChronoUnit unit) {
+            try {
+                Matcher matcher = matcher(value);
+                Assert.state(matcher.matches(), "Does not match simple duration pattern");
+                String suffix = matcher.group(2);
+                return (StringUtils.hasLength(suffix) ? Unit.fromSuffix(suffix) : Unit.fromChronoUnit(unit))
+                        .parse(matcher.group(1));
+            } catch (Exception ex) {
+                throw new IllegalArgumentException("'" + value + "' is not a valid simple duration", ex);
+            }
+        }
+
+        @Override
+        String print(Duration value, ChronoUnit unit) {
+            return Unit.fromChronoUnit(unit).print(value);
+        }
+
+    },
+
+    /**
+     * ISO-8601 formatting.
+     */
+    ISO8601("^[+-]?P.*$") {
+        @Override
+        Duration parse(String value, ChronoUnit unit) {
+            try {
+                return Duration.parse(value);
+            } catch (Exception ex) {
+                throw new IllegalArgumentException("'" + value + "' is not a valid ISO-8601 duration", ex);
+            }
+        }
+
+        @Override
+        String print(Duration value, ChronoUnit unit) {
+            return value.toString();
+        }
+
+    };
+
+    private final Pattern pattern;
+
+    DurationStyle(String pattern) {
+        this.pattern = Pattern.compile(pattern);
+    }
+
+    protected final boolean matches(String value) {
+        return this.pattern.matcher(value).matches();
+    }
+
+    protected final Matcher matcher(String value) {
+        return this.pattern.matcher(value);
+    }
+
+    /**
+     * Parse the given value to a duration.
+     * @param value the value to parse
+     * @return a duration
+     */
+    Duration parse(String value) {
+        return parse(value, null);
+    }
+
+    /**
+     * Parse the given value to a duration.
+     * @param value the value to parse
+     * @param unit the duration unit to use if the value doesn't specify one ({@code null}
+     * will default to ms)
+     * @return a duration
+     */
+    abstract Duration parse(String value, ChronoUnit unit);
```

Comment:
Could you create package-info.java that contains `NonNullByDefault`. 
And add
```suggestion
    abstract Duration parse(String value, @Nullable ChronoUnit unit);
```